base/applications/mspaint/winproc.cpp

```diff
@@ -575,6 +577,42 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+typedef ULONG (WINAPI *FN_MAPISendMail)(LHANDLE, ULONG_PTR, lpMapiMessage, FLAGS, ULONG);
+
+ULONG CMainWindow::OpenMailerWithAttachmentW(LPCWSTR pszFileName)
+{
+    if (!PathFileExists(pszFileName))
+        return MAPI_E_ATTACHMENT_NOT_FOUND;
+
+    MapiMessage message = { 0 };
+    message.lpszSubject = NULL;
+    if (pszFileName && *pszFileName)
+    {
+        char szFileA[MAX_PATH];
+        ::WideCharToMultiByte(CP_ACP, 0, pszFileName, -1, szFileA, _countof(szFileA), NULL, NULL);
+
+        MapiFileDesc attachment = { 0 };
+        attachment.nPosition = (ULONG)-1;
+        attachment.lpszPathName = szFileA;
+
+        message.nFileCount = 1;
+        message.lpFiles = &attachment;
+    }
+
+    ULONG result = MAPI_E_NOT_SUPPORTED;
+    HINSTANCE hMAPI32 = LoadLibraryW(L"mapi32.dll");
+    if (hMAPI32)
+    {
+        FN_MAPISendMail pMAPISendMail = (FN_MAPISendMail)GetProcAddress(hMAPI32, "MAPISendMail");
```

Comment:
Why not using the unicode function directly: MAPISendMailW ?
This would avoid using that inelegant explicit unicode to ansi conversion you do above.

Reply:
Fixed.